osquery/filesystem/linux/proc.cpp

```diff
@@ -16,48 +16,167 @@
 #include <osquery/filesystem.h>
 #include <osquery/logger.h>
 
+#include "osquery/core/conversions.h"
+#include "osquery/filesystem/linux/proc.h"
+
 namespace osquery {
+const std::vector<std::string> user_namespace_list = {
+    "cgroup", "ipc", "mnt", "net", "pid", "user", "uts"};
 
-const std::string kLinuxProcPath = "/proc";
+Status getNamespaceInode(ino_t& inode,
+                         const std::string& namespace_name,
+                         const std::string& path) {
+  inode = 0;
 
-Status procProcesses(std::set<std::string>& processes) {
-  // Iterate over each process-like directory in proc.
-  boost::filesystem::directory_iterator it(kLinuxProcPath), end;
-  try {
-    for (; it != end; ++it) {
-      if (boost::filesystem::is_directory(it->status())) {
-        // See #792: std::regex is incomplete until GCC 4.9
-        if (std::atoll(it->path().leaf().string().c_str()) > 0) {
-          processes.insert(it->path().leaf().string());
-        }
-      }
+  char link_destination[PATH_MAX] = {};
+
+  auto link_dest_length = readlink(path.data(), link_destination, PATH_MAX - 1);
```

Comment:
link_destination is zeroed because we have assigned an empty initializer list. When readlink writes to it, we will always have the bytes it hasn't touched set to 0

Reply:
Great! I didn't know the empty initialiser would zero the array.